framework/helpers/BaseStringHelper.php

```diff
@@ -54,46 +54,45 @@ public static function byteSubstr($string, $start, $length = null)
 
     /**
      * Returns the trailing name component of a path.
-     * This method is similar to the php function `basename()` except that it will
-     * treat both \ and / as directory separators, independent of the operating system.
-     * This method was mainly created to work on php namespaces. When working with real
-     * file paths, php's `basename()` should work fine for you.
+     * This method is similar to the php function `basename()` except that it will treat both
+     * `\` and `/` as directory separators, independent of the operating system. This method
+     * was mainly created to work on php namespaces. When working with real file paths, PHP's
+     * `basename()` should work fine for you.
      * Note: this method is not aware of the actual filesystem, or path components such as "..".
      *
      * @param string $path A path string.
      * @param string $suffix If the name component ends in suffix this will also be cut off.
-     * @return string the trailing name component of the given path.
-     * @see https://secure.php.net/manual/en/function.basename.php
+     * @return string The trailing name component of the given path.
+     * @see https://www.php.net/manual/en/function.basename
      */
     public static function basename($path, $suffix = '')
     {
-        $len = mb_strlen($suffix);
-        if ($len > 0 && mb_substr($path, -$len) === $suffix) {
-            $path = mb_substr($path, 0, -$len);
-        }
- 
-        $path = rtrim(str_replace('\\', '/', $path), '/');
-        $pos = mb_strrpos($path, '/');
-        if ($pos !== false) {
-            return mb_substr($path, $pos + 1);
+        if (!empty($suffix)) {
```

Comment:
`!empty($suffix)` is not the same as `mb_strlen($suffix) > 0` - it behaves differently if you pass `"0"` as `$suffix`.

Reply:
fixed